Some features/fixes

Fothsid opened the pull request:
P.S. Added feature to disable/enable logging because size of log after around 4 hours of playing was 400 MBs lol

Fothsid committed:
Add a simple CMakeLists.txt to CoreCLR wrapper.

Fothsid committed:
Add feature to disable/enable logging.

Fothsid committed:
Fix backslash addition to the name of HDD file.

Fothsid committed:
Make path to the x86 version of CoreCLR a little bit universal.

Fothsid committed:
Remove useless logging.

TheLastRar commented on CLR_DEV9/PSE/CLR_PSE_PluginLog.cs:
```diff
@@ -174,7 +175,13 @@ public static void Close()
+                if  ((!DEV9Header.config.EnableLogging.Error && eType == TraceEventType.Error) ||
+                    (!DEV9Header.config.EnableLogging.Verbose && eType == TraceEventType.Verbose) ||
+                    (!DEV9Header.config.EnableLogging.Information && eType == TraceEventType.Information))
+                    return;
```
Currently, logging is configured with `LogInit()` in DEV9.cs https://github.com/TheLastRar/CLR-DEV9/blob/2795c6e0455777d5dc38b6dd4cff2436d3b432b5/CLR_DEV9/DEV9.cs#L27 Which currently does so based on build type (Debug or Release)

Note that `LogInit()` is called before the configuration file have been loaded, I'm not sure what the best option to resolve this is.

You can call `SetStdOutLevel()`, `SetStdErrLevel()` & `SetFileLevel()` to configure global restrictions for each output. Any log levels set in `SetStdOutLevel()` must be excluded in`SetStdErrLevel()` or else you will get duplicate messages.

Also you must allow TraceEventType.Critical to log to stderr and the log file, this level is only used here in the event of a fatal error. https://github.com/TheLastRar/CLR-DEV9/blob/2795c6e0455777d5dc38b6dd4cff2436d3b432b5/CLR_DEV9/PSE/CLR_PSE_PluginLog.cs#L188

Fothsid replied:
Not really sure how I should handle this, because as you said, `LogInit()` is called before the configuration file have been loaded. 
When I made this, I was thinking only about how to stop these logs, I was not even thinking about build types, maybe because I can't find a way to select Release or Debug type in VS Code 0_o

TheLastRar replied:
I haven't used VSCode much myself, but I've made build tasks that should allow you to create release builds. (You will need to open the solution folder instead of the project folder is vscode to use the tasks)

You could either move any calls to `ConfigFile.LoadConf()` before any `LogInit()` (by moving `ConfigFile.LoadConf()` from `Open()` to `Init()` and reordering the calls in `Config()`) or by splitting `LogInit()` into two functions with Initialise logging with reasonable defaults before a second function is called to set logging levels as desired.

Ideally I would like to be able to enable Verbose logging on a per source level as just one source can produce a large amount entries, although the need for verbose logs doesn't come up often.

TheLastRar commented on CLR_DEV9/Config/ConfigLogging.cs:
```diff
@@ -0,0 +1,16 @@
+        [DataMember]
+        public bool Error = true;
+        [DataMember]
+        public bool Verbose = true;
```
Verbose should default to false to match release build behaviour.

TheLastRar commented on CLR_DEV9/Config/ConfigFile.cs:
```diff
@@ -47,6 +51,7 @@ private void Init()
             HddEnable = false;
             DirectConnectionSettings = new ConfigDirectIP();
             SocketConnectionSettings = new ConfigSocketIP();
+            EnableLogging = new ConfigLogging();
```
Do ConfigLogging's defaults apply correctly when loading an older config file (that lacks the entry)?

I remember having issues relating to that in the past.

Fothsid replied:
If config file lacks the entry, it applies default values. (at least on my machine)
However, if you manually add required entries, it still applies default values until you let it recreate config file. (maybe I just did it in wrong order?)

TheLastRar replied:
IIRC the entries in the file have to be in a specific order for settings to be read. I'm only concerned about updating an old file, and that seems to work.

TheLastRar commented on CLR_DEV9_LINUX/PSE.cpp:
```diff
@@ -303,8 +304,11 @@ void LoadCoreCLR(string pluginPath, string coreClrFolder)
+		if ( homedir == NULL ) {
+		    homedir = getpwuid(getuid())->pw_dir;
+		}
+		coreClrFolder = string(homedir) + string("/.config/PCSX2/coreclr");
```
CLR_DEV9_CORE.dll is currently found bases on the path passed to `DEV9setSettingsDir` while this assumes it's always in the home directory (it can be configured otherwise) Both should probably be consistent to avoid any confusion if a non-default settings path is encountered.

TheLastRar commented:
A Release build should disable most logging that ends up in the log file, although having this configurable would be nice

Fothsid committed:
Some changes.

Fothsid committed:
Use SDL2's Message Boxes instead of WinForms in .NET Core build.
Since SDL2 is one of the PCSX2's dependencies, I think this is a pretty good solution, at least for Linux
systems.
Should work on Windows too.

Fothsid committed:
- Moved all DEBUG build logging configurations before config file is loaded.
 - Moved all RELEASE build logging configurations after config file is loaded.
 - RELEASE build logging got enable/disable feature for each source source level through
config file

TheLastRar commented on CLR_DEV9/PSE/CLR_PSE_PluginLog.cs:
```diff
@@ -2,6 +2,7 @@
 using System.Collections.Generic;
 using System.Diagnostics;
 using System.IO;
+using CLRDEV9;
```
Remove the reference to CLRDEV9 as it's not needed.

TheLastRar commented on CLR_DEV9/PSE/CoreCLR/SDL2MessageBox.cs:
```diff
@@ -0,0 +1,26 @@
+            return SDL_ShowSimpleMessageBox((uint)flags, title, message, new IntPtr(0));
+        }
+    }
+}
```
nitpick: Newline at end of file

TheLastRar commented on CLR_DEV9_LINUX/PSE.cpp:
```diff
@@ -7,6 +7,7 @@
 #include <fcntl.h>
 #include <sys/types.h>
 #include <sys/stat.h>
+#include <pwd.h>
```
Remove reference if this is no-longer required.

Fothsid committed:
Remove references

Fothsid committed:
moo

TheLastRar reviewed (approved):
Looks good to me

Unless there is something else you want to add I will squash and merge this.

Fothsid commented:
That's all I wanted to add xD
